File: CopyPaste.Core/CardColor.cs
namespace CopyPaste.Core;

/// <summary>
/// Colors available for card visual organization.
/// Values are stored as integers in the database.
/// </summary>
public enum CardColor
{
    /// <summary>No color - default transparent border.</summary>
    None = 0,

    /// <summary>Red accent (#E74C3C).</summary>
    Red = 1,

    /// <summary>Green accent (#2ECC71).</summary>
    Green = 2,

    /// <summary>Purple accent (#9B59B6).</summary>
    Purple = 3,

    /// <summary>Yellow accent (#F1C40F).</summary>
    Yellow = 4,

    /// <summary>Blue accent (#3498DB).</summary>
    Blue = 5,

    /// <summary>Orange accent (#E67E22).</summary>
    Orange = 6
}

File: v2/core/lib/models/card_color.dart
enum CardColor {
  none(0, 0x00000000),
  red(1, 0xFFE74C3C),
  green(2, 0xFF2ECC71),
  purple(3, 0xFF9B59B6),
  yellow(4, 0xFFF1C40F),
  blue(5, 0xFF3498DB),
  orange(6, 0xFFE67E22);

  const CardColor(this.value, this.argb);

  final int value;
  final int argb;

  static CardColor fromValue(int value) =>
      CardColor.values.firstWhere((c) => c.value == value, orElse: () => none);
}
